src/dsymbol.h

```diff
@@ -279,6 +279,9 @@ class Dsymbol : public RootObject
     virtual AttribDeclaration *isAttribDeclaration() { return NULL; }
     virtual OverloadSet *isOverloadSet() { return NULL; }
     virtual void accept(Visitor *v) { v->visit(this); }
+    
+private:
+    Dsymbol *takeTypeTupleIndex(Loc loc, Scope *sc, Dsymbol *s, RootObject *id, Expression *expr);
```

Comment:
But error() is a member, so it's actually nicer to have this be a member, to have error messages as close as possible to the actual cause.